flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2338,82 +2338,129 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     //Needs PR Merge #3082
-    trySetThrustReductionAccelerationAltitude(s) {
+    async trySetThrustReductionAccelerationAltitude(s) {
         if (s === FMCMainDisplay.clrValue) {
             this.thrustReductionAltitudeIsPilotEntered = false;
             this.accelerationAltitudeIsPilotEntered = false;
-            CDUPerformancePage.UpdateThrRedAccFromOrigin(this, true, true);
+            CDUPerformancePage.UpdateThrRedAccFromOrigin(this);
             return true;
         }
+        if (s.length == 0 || s.length > 11) {
+            this.addNewMessage(NXSystemMessages.formatError);
+            return false;
+        }
+        if (s.length > 6 && !/^\d{4,5}\/\d{4,5}$/.test(s)) {
+            this.addNewMessage(NXSystemMessages.formatError);
+            return false;
+        }
```

Comment:
Deal with the whole formatting issue with the one regex... job done.
```js
        const [match, thrRedAlt, slash1, , slash2, accAlt] = s.match(/^([0-9]{4,5})?(\/?)?((\/)([0-9]{4,5}))?$/) || [];
        if (!match || (slash1 && slash2) || (!thrRedAlt && !accAlt)) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }
        
        // use thrRedAlt and accAlt
        // check if undefined first, which one will be if only one was entered
        // if not undefined parseInt is guaranteed to work because the regex already verified their format
```

Reply:
Thanks this looks very neat. I will put this in and test all likely pilot inputs against the code later today. Then update the PR.

Reply:
Lovely code for checking the format, thanks.  I changed the thrRedAlt and accAlt to "checkThrRedAlt" & "checkAccAlt". This is so match is kept a const.  Then kept thrRedAlt and accAlt to store and change value as needed.
All the modifications have been check for the values and errors that I think would be input.